src/test/java/htsjdk/TestDataProviders.java

```diff
@@ -0,0 +1,75 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.NoInjection;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This test is a mechanism to check that none of the data-providers fail to run.
+ * It is needed because in the case that a data-provider fails (for some reason, perhaps a change in some other code
+ * causes it to throw an exception) the tests that rely on it will be silently skipped.
+ * The only mention of this will be in test logs but since we normally avoid reading these logs and rely on the
+ * exit code, it will look like all the tests have passed.
+ *
+ * @author Yossi Farjoun
+ */
+public class TestDataProviders {
+
+
+    @Test
+    public void IndependentTestOfDataProviderTest() throws IllegalAccessException, InvocationTargetException, InstantiationException {
+        testAllDataProvidersdata();
+    }
+
+    @DataProvider(name = "DataprovidersThatDontTestThemselves")
+    public Iterator<Object[]> testAllDataProvidersdata() throws IllegalAccessException, InstantiationException, InvocationTargetException {
+
+        List<Object[]> data = new ArrayList<>();
+        final ClassFinder classFinder = new ClassFinder();
+        classFinder.find("htsjdk", Object.class);
+
+        for (final Class<?> testClass : classFinder.getClasses()) {
+            if (Modifier.isAbstract(testClass.getModifiers())) continue;
+            for (final Method method : testClass.getMethods()) {
+                if (method.isAnnotationPresent(DataProvider.class)) {
+                    data.add(new Object[]{method, testClass});
+                }
+            }
+        }
+        Assert.assertTrue(data.size() > 1);
+
+        // make sure that this @DataProvider is in the list
+        Assert.assertEquals(data.stream().filter(c -> ((Method) c[0]).getName().equals("testAllDataProvidersdata")).count(), 1);
+
+        return data.iterator();
+    }
+
+    // @NoInjection annotations required according to this test:
+    // https://github.com/cbeust/testng/blob/master/src/test/java/test/inject/NoInjectionTest.java
+    @Test(dataProvider = "DataprovidersThatDontTestThemselves")
+    public void testDataProviderswithDP(@NoInjection final Method method, final Class clazz) throws IllegalAccessException, InstantiationException, InvocationTargetException {
+
+        System.err.println("Method: " + method + " Class: " + clazz.getName());
+
+        Object instance = clazz.newInstance();
+
+        // Some tests assume that the @BeforeSuite methods will be called before the @DataProviders
+        for (final Method otherMethod : clazz.getMethods()) {
+            if (otherMethod.isAnnotationPresent(BeforeSuite.class)) {
+                otherMethod.invoke(instance);
+            }
+        }
+
+        method.invoke(instance);
```

Comment:
I think we should make sure that if this throws it has a very clear error message which explains what this test is, what exact DataProvider it was calling, and what the exception thrown was.  Otherwise people are going to be mystified by it.

Reply:
ok. I'll see what I can do.